src/states/node.rs

```diff
@@ -168,38 +171,26 @@ impl Node {
         }
     }
 
-    #[allow(clippy::too_many_arguments)]
     pub fn from_establishing_node(
-        ack_mgr: AckManager,
-        cache: Box<Cache>,
-        chain: Chain,
-        crust_service: Service,
-        full_id: FullId,
-        gen_pfx_info: GenesisPfxInfo,
-        msg_queue: VecDeque<RoutingMessage>,
-        notified_nodes: BTreeSet<PublicId>,
-        old_pfx: Prefix<XorName>,
-        parsec_map: ParsecMap,
-        peer_mgr: PeerManager,
-        routing_msg_filter: RoutingMessageFilter,
+        source: EstablishingNode,
         sec_info: SectionInfo,
-        timer: Timer,
+        old_pfx: Prefix<XorName>,
         outbox: &mut EventBox,
     ) -> Result<Self, RoutingError> {
         let mut node = Self::new(
-            ack_mgr,
-            cache,
-            chain,
-            crust_service,
-            full_id,
-            gen_pfx_info,
+            source.ack_mgr,
+            source.cache,
+            source.chain,
+            source.crust_service,
+            source.full_id,
+            source.gen_pfx_info,
             false,
-            msg_queue,
-            notified_nodes,
-            parsec_map,
-            peer_mgr,
-            routing_msg_filter,
-            timer,
+            source.msg_backlog.into_iter().collect(),
```

Comment:
Kind of lost the benefit here because there is a new function.

So we would need to pass the source to new if possible, otherwise that is the flexibility we lost that we had with all the parameter, that would still be there with pretty much no additional noise with a separate struct (i.e the parameter are just moved to the struct, same amount of code).

I quite like your approach, because it remove the need for the argument to be described again.

However, it create some coupling and force you to make these member public.
I'm thinking it could be cleaner with a separate struct as it would not have these quirks.

Reply:
This is a follow up on resolution for https://github.com/maidsafe/routing/pull/1616#pullrequestreview-230608692.

Could you comment on that.

Reply:
I'm OK to change the code to use separate structs if you think it's worth it. But I'm also OK with keeping it as is, because I don't think the additional coupling is too bad in practice and it makes the code simpler. Your call.

Reply:
I think it is worth it, it will be cleaner and mirror simply the interface that existed before.
But mostly I'm wanting to fix this `new` function that this comment is on as it push around the problem we were trying to solve (i.e `new` need to take the struct, so names are tied together).